packages/wallets/src/vultisig/walletHelpers.ts

```diff
@@ -0,0 +1,368 @@
+      abi,
+      funcName,
+      funcParams = [],
+      txOverrides,
+    }: CallParams): Promise<T> => {
+      if (!contractAddress) {
+        throw new SwapKitError("wallet_vultisig_contract_address_not_provided");
+      }
+      const { createContract, getCreateContractTxObject, isStateChangingCall } = await import(
+        "@swapkit/toolboxes/evm"
+      );
+
+      const isStateChanging = isStateChangingCall({ abi, funcName });
+
+      if (isStateChanging) {
+        const createTx = getCreateContractTxObject({ provider, chain });
+        const { value, from, to, data } = await createTx({
+          contractAddress,
+          abi,
+          funcName,
+          funcParams,
+          txOverrides,
+        });
+
+        const signer = await provider.getSigner();
+        const tx = await signer.sendTransaction({
+          value: BigInt(value || 0),
+          from,
+          to,
+          data: data || "0x",
+        });
+        return tx.hash as T;
+      }
+      const contract = createContract(contractAddress, abi, provider);
+
+      const result = await contract[funcName]?.(...funcParams);
+
+      return typeof result?.hash === "string" ? result?.hash : result;
+    },
+    approve: async ({ assetAddress, spenderAddress, amount, from }: ApproveParams) => {
+      const { MAX_APPROVAL, getCreateContractTxObject } = await import("@swapkit/toolboxes/evm");
+      const funcParams = [spenderAddress, BigInt(amount || MAX_APPROVAL)];
+      const txOverrides = { from };
+
+      const functionCallParams = {
+        contractAddress: assetAddress,
+        abi: erc20ABI,
+        funcName: "approve",
+        funcParams,
+        txOverrides,
+      };
+
+      const createTx = getCreateContractTxObject({ provider, chain });
+      const { value, to, data } = await createTx(functionCallParams);
+
+      const signer = await provider.getSigner();
+      const tx = await signer.sendTransaction({
+        value: BigInt(value || 0),
+        from,
+        to,
+        data: data || "0x",
+      });
+      return tx.hash;
+    },
+    sendTransaction: async (txParams: EVMTxParams) => {
+      const { from, to, data, value } = txParams;
+      if (!to) {
+        throw new SwapKitError("wallet_vultisig_send_transaction_no_address");
+      }
+
+      const signer = await provider.getSigner();
+      const tx = await signer.sendTransaction({
+        value: BigInt(value || 0),
+        from,
+        to,
+        data: data || "0x",
+      });
+      return tx.hash;
+    },
+  };
+}
+
+export async function switchCosmosWalletNetwork(
+  provider: VultisigCosmosProvider,
+  chain: Chain.Cosmos | Chain.Kujira,
+  networkParams?: NetworkParams,
+) {
+  try {
+    await provider.request({
+      method: "wallet_switch_chain",
+      params: [{ chainId: ChainToChainId[chain] }],
+    });
+  } catch (_error) {
+    if (!networkParams) {
+      throw new SwapKitError("helpers_failed_to_switch_network", {
+        error: _error,
```

Comment:
We're using `_` prefix only for unused variables :)

```suggestion
  } catch (error) {
    if (!networkParams) {
      throw new SwapKitError("helpers_failed_to_switch_network", {
        error,
```